cob: fix panic on m0__obj_namei_send()

andriytk opened the pull request:
When the pool width is less than the number of IOS-services
configured (which may happen in a multi-tier setups), and
`md_redundancy` is bigger than the pool width, we get
SIGSEGV at `cob_ios_md_send()`:

    motr[48789]:  7600  FATAL  [lib/assert.c:50:m0_panic]  panic: fatal signal delivered at unknown() (unknown:0)  [git: 2.0.0-307-11-gcacd425] /tmp/m0trace.48789
    Motr panic: fatal signal delivered at unknown() unknown:0 (errno: 4) (last failed: none) [git: 2.0.0-307-11-gcacd425] pid: 48789  /tmp/m0trace.48789
    Motr panic reason: signo: 11
    /lib64/libmotr.so.2(m0_arch_backtrace+0x2f)[0x7f79535f840f]
    /lib64/libmotr.so.2(m0_arch_panic+0xf3)[0x7f79535f85f3]
    /lib64/libmotr.so.2(+0x353684)[0x7f79535e8684]
    /lib64/libmotr.so.2(+0x363638)[0x7f79535f8638]
    /lib64/libpthread.so.0(+0xf630)[0x7f7952d86630]
    /lib64/libmotr.so.2(+0x392ed5)[0x7f7953627ed5]
    /lib64/libmotr.so.2(+0x39337d)[0x7f795362837d]
    /lib64/libmotr.so.2(+0x393686)[0x7f7953628686]
    /lib64/libmotr.so.2(m0__obj_namei_send+0x15a)[0x7f7953628e6a]
    /lib64/libmotr.so.2(+0x394c72)[0x7f7953629c72]
    /lib64/libmotr.so.2(m0_op_launch_one+0x12a)[0x7f795360f1ea]
    /lib64/libmotr.so.2(m0_op_launch+0x54)[0x7f795360f344]

    (gdb) l *(+0x393686)
    0x393686 is in cob_req_send (motr/cob.c:1427).
    1422		 * race condition!
    1423		 */
    1424		cr->cr_icr_nr = cinst->m0c_pools_common.pc_md_redundancy;
    1425		cr->cr_cob_type = M0_COB_MD;
    1426		rc = (cr->cr_flags & COB_REQ_SYNC) ?
    1427		     cob_ios_req_send_sync(cr) :
    1428		     cob_ios_req_send_async(cr);
    1429
    1430		return M0_RC(rc);
    1431	}
    (gdb) l *(+0x39337d)
    0x39337d is in cob_ios_req_send_async (motr/cob.c:1320).
    1315
    1316		/*
    1317		 * Prepare all fops before sending. It avoids race between
    1318		 * setting cr->cr_ios_fop and icr_ast() execution.
    1319		 */
    1320		for (i = 0; i < icr_nr; ++i) {
    1321			rc = cob_ios_prepare(cr, i);
    1322			/*
    1323			 * Silence possible compiler's warnings about re-assigning rc.
    1324			 * If prepare phase fails for all fops, we will return the last
    (gdb) l *(+0x392ed5)
    0x392ed5 is in cob_ios_prepare (motr/cob.c:1065).
    1060			 * then allocated.
    1061			 *
    1062			 * TODO: revisit to check if we can re-use fop.
    1063			 */
    1064			fop = cr->cr_ios_fop[i];
    1065			M0_ASSERT(fop->f_opaque != NULL);
    1066
    1067			cob_ios_fop_fini(fop);
    1068			cr->cr_ios_fop[i] = NULL;
    1069		}

The reason is that we try to send more md-fops than was
allocated at `cob_req_alloc()`.

Solution: make sure at `cob_ios_md_send()` that `md_redundancy`
does not exceed the pool width.

andriytk commented:
The patch has been successfully tested on the Sage prototype cluster (with 7 IOS-services and pool width 4):

```
$ export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/home/users/jusers/tkachuk1/sage/motr/motr/.libs/
$ m0cp -p 0x7000000000000001:0x531 -H 172.18.1.1@o2ib:12345:1:1 -l 172.18.1.1@o2ib:12345:4:1 -P 0x7200000000000001:0x19 -o 0x12340506001:0x78632001 -L 1 -s 32k -c 1 /tmp/32k 
$ 
```

andriytk commented:
Started http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767/

huanghua78 commented:
Please check the following code:
```
1352 static void cob_ast_ios_io_send(struct m0_sm_group *grp,
1353                                 struct m0_sm_ast *ast)
1354 {
1355         int                      rc;
...
...
1382
1383         /* Send a fop to each COB. */
1384         cr->cr_cob_type = M0_COB_IO;
1385         cr->cr_icr_nr = pool_width;
1386         rc = cob_ios_req_send_async(cr);
```

Is this assignment of "cr->cr_icr_nr" on L1385 still valid?

andriytk commented:
Good catch @huanghua78! Fixed. Thanks.

andriytk commented:
Started http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/770/

cortx-admin commented:
<h1>Jenkins CI Result : <a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767/>Motr#767</a></h1><h2>Motr Test Summary</h2><table><tr><th>Test Result</th><th>Count</th><th>Info</tr><tr><td>:x:Failed</td><td>8</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/01motr-single-node/28motr-sys-kvs-test target=_blank>01motr-single-node/28motr-sys-kvs-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/01motr-single-node/46m0crate target=_blank>01motr-single-node/46m0crate</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/01motr-single-node/52singlenode-sanity target=_blank>01motr-single-node/52singlenode-sanity</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/03motr-single-node/56sns-repair-motr-mf target=_blank>03motr-single-node/56sns-repair-motr-mf</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/04motr-single-node/49motr-rpc-cancel target=_blank>04motr-single-node/49motr-rpc-cancel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/04motr-single-node/50motr-rm-lock-cc-io target=_blank>04motr-single-node/50motr-rm-lock-cc-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/04motr-single-node/51motr-rmw target=_blank>04motr-single-node/51motr-rmw</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/05motr-single-node/42motr-utils target=_blank>05motr-single-node/42motr-utils</a></p></details></td></tr><tr><td>:checkered_flag:Skipped</td><td>5</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/01motr-single-node/37protocol target=_blank>01motr-single-node/37protocol</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/04motr-single-node/31motr-sys-kvs-kernel-test target=_blank>04motr-single-node/31motr-sys-kvs-kernel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/05motr-single-node/07multi-client-tests target=_blank>05motr-single-node/07multi-client-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/05motr-single-node/12sns-cc-io-tests target=_blank>05motr-single-node/12sns-cc-io-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/05motr-single-node/39mount-fail target=_blank>05motr-single-node/39mount-fail</a></p></details></td></tr><tr><td>:heavy_check_mark:Passed</td><td>51</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/01motr-single-node/00userspace-tests target=_blank>01motr-single-node/00userspace-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/01motr-single-node/01kernel-tests target=_blank>01motr-single-node/01kernel-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/01motr-single-node/03initscript-tests target=_blank>01motr-single-node/03initscript-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/01motr-single-node/04net-tests target=_blank>01motr-single-node/04net-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/01motr-single-node/05rpcping-tests target=_blank>01motr-single-node/05rpcping-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/01motr-single-node/06console-tests target=_blank>01motr-single-node/06console-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/01motr-single-node/29confgen-test target=_blank>01motr-single-node/29confgen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/01motr-single-node/30hagen-test target=_blank>01motr-single-node/30hagen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/01motr-single-node/35m0singlenode target=_blank>01motr-single-node/35m0singlenode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/02motr-single-node/02system-tests target=_blank>02motr-single-node/02system-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/02motr-single-node/08degraded-mode-tests target=_blank>02motr-single-node/08degraded-mode-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/02motr-single-node/09poolmach-tests target=_blank>02motr-single-node/09poolmach-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/02motr-single-node/20pool-version-test target=_blank>02motr-single-node/20pool-version-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/02motr-single-node/21spiel-test target=_blank>02motr-single-node/21spiel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/02motr-single-node/22rpc-cancel-test target=_blank>02motr-single-node/22rpc-cancel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/02motr-single-node/23m0d-signal-test target=_blank>02motr-single-node/23m0d-signal-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/02motr-single-node/24m0d-fsync-test target=_blank>02motr-single-node/24m0d-fsync-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/02motr-single-node/41motr-conf-update target=_blank>02motr-single-node/41motr-conf-update</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/02motr-single-node/47motr-tests-user-kernel target=_blank>02motr-single-node/47motr-tests-user-kernel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/02motr-single-node/51kem target=_blank>02motr-single-node/51kem</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/02motr-single-node/53clusterusage-alert target=_blank>02motr-single-node/53clusterusage-alert</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/03motr-single-node/16conf-st-test target=_blank>03motr-single-node/16conf-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/03motr-single-node/17sss-st-test target=_blank>03motr-single-node/17sss-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/03motr-single-node/36spare-reservation target=_blank>03motr-single-node/36spare-reservation</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/03motr-single-node/54sns-repair-motr-1f target=_blank>03motr-single-node/54sns-repair-motr-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/03motr-single-node/55sns-repair-motr-1n-1f target=_blank>03motr-single-node/55sns-repair-motr-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/03motr-single-node/57sns-repair-motr-1k-1f target=_blank>03motr-single-node/57sns-repair-motr-1k-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/03motr-single-node/58sns-repair-motr-ios-fail target=_blank>03motr-single-node/58sns-repair-motr-ios-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/03motr-single-node/59sns-repair-motr-abort target=_blank>03motr-single-node/59sns-repair-motr-abort</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/03motr-single-node/60sns-repair-motr-abort-quiesce target=_blank>03motr-single-node/60sns-repair-motr-abort-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/03motr-single-node/61spiel-multi-confd target=_blank>03motr-single-node/61spiel-multi-confd</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/04motr-single-node/19sns-abort-test target=_blank>04motr-single-node/19sns-abort-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/04motr-single-node/25spiel-sns-repair-test target=_blank>04motr-single-node/25spiel-sns-repair-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/04motr-single-node/26spiel-sns-repair-quiesce-test target=_blank>04motr-single-node/26spiel-sns-repair-quiesce-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/04motr-single-node/27sns-repair-io-fail-test target=_blank>04motr-single-node/27sns-repair-io-fail-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/04motr-single-node/32m0t1fs-rconfc-fail-test target=_blank>04motr-single-node/32m0t1fs-rconfc-fail-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/04motr-single-node/33motr-st target=_blank>04motr-single-node/33motr-st</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/04motr-single-node/34sns-repair-1n-1f target=_blank>04motr-single-node/34sns-repair-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/04motr-single-node/48motr-raid0-io target=_blank>04motr-single-node/48motr-raid0-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/05motr-single-node/10sns-single-tests target=_blank>05motr-single-node/10sns-single-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/05motr-single-node/11sns-multi-tests target=_blank>05motr-single-node/11sns-multi-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/05motr-single-node/13sns-repair-quiesce target=_blank>05motr-single-node/13sns-repair-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/05motr-single-node/14m0t1fs-fsync-test target=_blank>05motr-single-node/14m0t1fs-fsync-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/05motr-single-node/15m0t1fs-fwait-test target=_blank>05motr-single-node/15m0t1fs-fwait-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/05motr-single-node/18m0mount-test target=_blank>05motr-single-node/18m0mount-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/05motr-single-node/38sns-abort-quiesce-test target=_blank>05motr-single-node/38sns-abort-quiesce-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/05motr-single-node/40motr-dgmode target=_blank>05motr-single-node/40motr-dgmode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/05motr-single-node/43motr-sync-replication target=_blank>05motr-single-node/43motr-sync-replication</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/05motr-single-node/44motr-sns-repair target=_blank>05motr-single-node/44motr-sns-repair</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/05motr-single-node/45motr-sns-repair-N-1 target=_blank>05motr-single-node/45motr-sns-repair-N-1</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit/motr-single-node/62dix-repair-st-lookup-insert target=_blank>motr-single-node/62dix-repair-st-lookup-insert</a></p></details></td></tr><tr><td>Total</td><td>64</td><td><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/767//testReport/junit>:link:</a></td></tr></table><h2> CppCheck Summary</h2><p> &emsp;&emsp;&emsp;Cppcheck: No new warnings found :thumbsup:</p>

andriytk commented:
Started http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771/

cortx-admin commented:
<h1>Jenkins CI Result : <a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771/>Motr#771</a></h1><h2>Motr Test Summary</h2><table><tr><th>Test Result</th><th>Count</th><th>Info</tr><tr><td>:x:Failed</td><td>4</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/02motr-single-node/23m0d-signal-test target=_blank>02motr-single-node/23m0d-signal-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/02motr-single-node/41motr-conf-update target=_blank>02motr-single-node/41motr-conf-update</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/04motr-single-node/51motr-rmw target=_blank>04motr-single-node/51motr-rmw</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/01motr-single-node/00userspace-tests target=_blank>01motr-single-node/00userspace-tests</a></p></details></td></tr><tr><td>:checkered_flag:Skipped</td><td>5</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/01motr-single-node/37protocol target=_blank>01motr-single-node/37protocol</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/04motr-single-node/31motr-sys-kvs-kernel-test target=_blank>04motr-single-node/31motr-sys-kvs-kernel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/05motr-single-node/07multi-client-tests target=_blank>05motr-single-node/07multi-client-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/05motr-single-node/12sns-cc-io-tests target=_blank>05motr-single-node/12sns-cc-io-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/05motr-single-node/39mount-fail target=_blank>05motr-single-node/39mount-fail</a></p></details></td></tr><tr><td>:heavy_check_mark:Passed</td><td>55</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/01motr-single-node/01kernel-tests target=_blank>01motr-single-node/01kernel-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/01motr-single-node/03initscript-tests target=_blank>01motr-single-node/03initscript-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/01motr-single-node/04net-tests target=_blank>01motr-single-node/04net-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/01motr-single-node/05rpcping-tests target=_blank>01motr-single-node/05rpcping-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/01motr-single-node/06console-tests target=_blank>01motr-single-node/06console-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/01motr-single-node/28motr-sys-kvs-test target=_blank>01motr-single-node/28motr-sys-kvs-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/01motr-single-node/29confgen-test target=_blank>01motr-single-node/29confgen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/01motr-single-node/30hagen-test target=_blank>01motr-single-node/30hagen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/01motr-single-node/35m0singlenode target=_blank>01motr-single-node/35m0singlenode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/01motr-single-node/46m0crate target=_blank>01motr-single-node/46m0crate</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/01motr-single-node/52singlenode-sanity target=_blank>01motr-single-node/52singlenode-sanity</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/02motr-single-node/02system-tests target=_blank>02motr-single-node/02system-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/02motr-single-node/08degraded-mode-tests target=_blank>02motr-single-node/08degraded-mode-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/02motr-single-node/09poolmach-tests target=_blank>02motr-single-node/09poolmach-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/02motr-single-node/20pool-version-test target=_blank>02motr-single-node/20pool-version-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/02motr-single-node/21spiel-test target=_blank>02motr-single-node/21spiel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/02motr-single-node/22rpc-cancel-test target=_blank>02motr-single-node/22rpc-cancel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/02motr-single-node/24m0d-fsync-test target=_blank>02motr-single-node/24m0d-fsync-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/02motr-single-node/47motr-tests-user-kernel target=_blank>02motr-single-node/47motr-tests-user-kernel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/02motr-single-node/51kem target=_blank>02motr-single-node/51kem</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/02motr-single-node/53clusterusage-alert target=_blank>02motr-single-node/53clusterusage-alert</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/03motr-single-node/16conf-st-test target=_blank>03motr-single-node/16conf-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/03motr-single-node/17sss-st-test target=_blank>03motr-single-node/17sss-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/03motr-single-node/36spare-reservation target=_blank>03motr-single-node/36spare-reservation</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/03motr-single-node/54sns-repair-motr-1f target=_blank>03motr-single-node/54sns-repair-motr-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/03motr-single-node/55sns-repair-motr-1n-1f target=_blank>03motr-single-node/55sns-repair-motr-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/03motr-single-node/56sns-repair-motr-mf target=_blank>03motr-single-node/56sns-repair-motr-mf</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/03motr-single-node/57sns-repair-motr-1k-1f target=_blank>03motr-single-node/57sns-repair-motr-1k-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/03motr-single-node/58sns-repair-motr-ios-fail target=_blank>03motr-single-node/58sns-repair-motr-ios-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/03motr-single-node/59sns-repair-motr-abort target=_blank>03motr-single-node/59sns-repair-motr-abort</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/03motr-single-node/60sns-repair-motr-abort-quiesce target=_blank>03motr-single-node/60sns-repair-motr-abort-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/03motr-single-node/61spiel-multi-confd target=_blank>03motr-single-node/61spiel-multi-confd</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/04motr-single-node/19sns-abort-test target=_blank>04motr-single-node/19sns-abort-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/04motr-single-node/25spiel-sns-repair-test target=_blank>04motr-single-node/25spiel-sns-repair-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/04motr-single-node/26spiel-sns-repair-quiesce-test target=_blank>04motr-single-node/26spiel-sns-repair-quiesce-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/04motr-single-node/27sns-repair-io-fail-test target=_blank>04motr-single-node/27sns-repair-io-fail-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/04motr-single-node/32m0t1fs-rconfc-fail-test target=_blank>04motr-single-node/32m0t1fs-rconfc-fail-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/04motr-single-node/33motr-st target=_blank>04motr-single-node/33motr-st</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/04motr-single-node/34sns-repair-1n-1f target=_blank>04motr-single-node/34sns-repair-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/04motr-single-node/48motr-raid0-io target=_blank>04motr-single-node/48motr-raid0-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/04motr-single-node/49motr-rpc-cancel target=_blank>04motr-single-node/49motr-rpc-cancel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/04motr-single-node/50motr-rm-lock-cc-io target=_blank>04motr-single-node/50motr-rm-lock-cc-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/05motr-single-node/10sns-single-tests target=_blank>05motr-single-node/10sns-single-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/05motr-single-node/11sns-multi-tests target=_blank>05motr-single-node/11sns-multi-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/05motr-single-node/13sns-repair-quiesce target=_blank>05motr-single-node/13sns-repair-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/05motr-single-node/14m0t1fs-fsync-test target=_blank>05motr-single-node/14m0t1fs-fsync-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/05motr-single-node/15m0t1fs-fwait-test target=_blank>05motr-single-node/15m0t1fs-fwait-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/05motr-single-node/18m0mount-test target=_blank>05motr-single-node/18m0mount-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/05motr-single-node/38sns-abort-quiesce-test target=_blank>05motr-single-node/38sns-abort-quiesce-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/05motr-single-node/40motr-dgmode target=_blank>05motr-single-node/40motr-dgmode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/05motr-single-node/42motr-utils target=_blank>05motr-single-node/42motr-utils</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/05motr-single-node/43motr-sync-replication target=_blank>05motr-single-node/43motr-sync-replication</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/05motr-single-node/44motr-sns-repair target=_blank>05motr-single-node/44motr-sns-repair</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/05motr-single-node/45motr-sns-repair-N-1 target=_blank>05motr-single-node/45motr-sns-repair-N-1</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit/motr-single-node/62dix-repair-st-lookup-insert target=_blank>motr-single-node/62dix-repair-st-lookup-insert</a></p></details></td></tr><tr><td>Total</td><td>64</td><td><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/771//testReport/junit>:link:</a></td></tr></table><h2> CppCheck Summary</h2><p> &emsp;&emsp;&emsp;Cppcheck: No new warnings found :thumbsup:</p>

andriytk commented:
The original patch was wrong - it did not take into account that we do need to send fops to all cobs in the pool sometimes. So I reverted most of it back and changed the solution to check for the pool_width only at `cob_ios_md_send()`.

cortx-admin commented:
<h1>Jenkins CI Result : <a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/772/>Motr#772</a></h1><h2>Motr Test Summary</h2><table><tr><th>Test Result</th><th>Count</th><th>Info</tr><tr><td>:x:Failed</td><td>1</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/772//testReport/junit/01motr-single-node/00userspace-tests target=_blank>01motr-single-node/00userspace-tests</a></p></details></td></tr><tr><td>:checkered_flag:Skipped</td><td>1</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/772//testReport/junit/01motr-single-node/37protocol target=_blank>01motr-single-node/37protocol</a></p></details></td></tr><tr><td>:heavy_check_mark:Passed</td><td>11</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/772//testReport/junit/01motr-single-node/01kernel-tests target=_blank>01motr-single-node/01kernel-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/772//testReport/junit/01motr-single-node/03initscript-tests target=_blank>01motr-single-node/03initscript-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/772//testReport/junit/01motr-single-node/04net-tests target=_blank>01motr-single-node/04net-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/772//testReport/junit/01motr-single-node/05rpcping-tests target=_blank>01motr-single-node/05rpcping-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/772//testReport/junit/01motr-single-node/06console-tests target=_blank>01motr-single-node/06console-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/772//testReport/junit/01motr-single-node/28motr-sys-kvs-test target=_blank>01motr-single-node/28motr-sys-kvs-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/772//testReport/junit/01motr-single-node/29confgen-test target=_blank>01motr-single-node/29confgen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/772//testReport/junit/01motr-single-node/30hagen-test target=_blank>01motr-single-node/30hagen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/772//testReport/junit/01motr-single-node/35m0singlenode target=_blank>01motr-single-node/35m0singlenode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/772//testReport/junit/01motr-single-node/46m0crate target=_blank>01motr-single-node/46m0crate</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/772//testReport/junit/01motr-single-node/52singlenode-sanity target=_blank>01motr-single-node/52singlenode-sanity</a></p></details></td></tr><tr><td>Total</td><td>13</td><td><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/772//testReport/junit>:link:</a></td></tr></table><h2> CppCheck Summary</h2><p> &emsp;&emsp;&emsp;Cppcheck: No new warnings found :thumbsup:</p>

andriytk commented:
Started http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773/

mehjoshi commented:
requesting @nikitadanilov and @madhavemuri to re-review based on the updated changes.

madhavemuri commented:
@andriytk : This can cause internal meta-data unavailability for large pool users, in cases when number of node failures to be supported is greater than the pool width of small pool. I think we need to use a base pool with large number of ioservice/pool width only in case of multiple pools.
Can you share confd.xc and Cluster.yaml from your setup ?

andriytk commented:
@madhavemuri, in the Sage cluster there are 7 server nodes (each having IOS configured) and many pools:

```
Data pools:
    # fid name
    0x6f00000000000001:0x47d 'tier1-nvme-p1'
    0x6f00000000000001:0x48a 'tier2-ssd-p2'
    0x6f00000000000001:0x497 'tier3-hdd-p3'
    0x6f00000000000001:0x4a2 'tier2-ssd'
    0x6f00000000000001:0x4bf 'tier3-hdd'
```

The first three pools above have only 4 storage devices configured. (All the rest storage devices are assigned to the remaining pools.) See [confd.xc](https://github.com/Seagate/cortx-motr/files/7236108/confd.xc.txt) and [sage-proto-cdf.yaml](https://github.com/Seagate/cortx-motr/files/7236109/sage-proto-cdf.yaml.txt) files attached.

madhavemuri commented:
I see that in `sage-proto-cdf.yaml` node failures are not configured, above issue is relevant when node failure domain is used.

Yes, mdpool will need to have the failure domains compared to the large data pool in the cluster and md redundancy can be upto K of that pool..

    # cat confd.cg | grep pool-0x518 -A1
(root-0 verno=1 rootfid=root-0 mdpool=**pool-0x518** imeta_pver=pver-0x500 **mdredundancy=7** params=[] nodes=[node-3, node-31, node-56, node-83, node-0x76, node-0x98, node-0xb9, node-0x12d, node-0x162, node-0x197, node-0x1cc, node-0x201, node-0x236, node-0x26b, node-0x2a0, node-0x2d5, node-0x30a, node-0x33f, node-0x374, node-0x3a9, node-0x3de, node-0x413, node-0x448] sites=[site-1] pools=[pool-0x47d, pool-0x48a, pool-0x497, pool-0x4a2, pool-0x4bf, pool-0x4f1, pool-0x518] profiles=[profile-0x531] fdmi_flt_grps=[])
(site-1 racks=[rack-2] pvers=[pver-0x47e, pver-0x48b, pver-0x498, pver-0x4a3, pver-0x4c0, pver-0x500, pver-0x519])
--
(pool-0x518 pver_policy=0 pvers=[pver-0x519])
**(pver-0x519 N=1 K=6 S=0 P=7 tolerance=[0, 0, 6, 6, 6] sitevs=[objv-0x51e])**
--
(profile-0x531 pools=[pool-0x47d, pool-0x48a, pool-0x497, pool-0x4a2, pool-0x4bf, pool-0x4f1, pool-0x518])
]# cat confd.cg | grep toler
(pver-0x47e N=2 K=1 S=1 P=4 tolerance=[0, 0, 0, 0, 1] sitevs=[objv-0x483])
(pver-0x48b N=2 K=1 S=1 P=4 tolerance=[0, 0, 0, 0, 1] sitevs=[objv-0x490])
(pver-0x498 N=2 K=1 S=1 P=4 tolerance=[0, 0, 0, 0, 1] sitevs=[objv-0x49d])
(pver-0x4a3 N=8 K=2 S=2 P=15 tolerance=[0, 0, 0, 0, 2] sitevs=[objv-0x4a8])
(pver-0x4c0 N=8 K=2 S=2 P=42 tolerance=[0, 0, 0, 0, 2] sitevs=[objv-0x4c5])
(pver-0x500 N=1 K=6 S=0 P=7 tolerance=[0, 0, 6, 6, 6] sitevs=[objv-0x505])
(pver-0x519 N=1 K=6 S=0 P=7 tolerance=[0, 0, 6, 6, 6] sitevs=[objv-0x51e])

I think we need a configurable option or generic way to calculate the value for md_redundancy.
In this case it needs to be equal to largest K of data pools, i.e 2 instead of 7.

cortx-admin commented:
<h1>Jenkins CI Result : <a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773/>Motr#773</a></h1><h2>Motr Test Summary</h2><table><tr><th>Test Result</th><th>Count</th><th>Info</tr><tr><td>:x:Failed</td><td>2</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/03motr-single-node/16conf-st-test target=_blank>03motr-single-node/16conf-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/01motr-single-node/00userspace-tests target=_blank>01motr-single-node/00userspace-tests</a></p></details></td></tr><tr><td>:checkered_flag:Skipped</td><td>5</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/01motr-single-node/37protocol target=_blank>01motr-single-node/37protocol</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/04motr-single-node/31motr-sys-kvs-kernel-test target=_blank>04motr-single-node/31motr-sys-kvs-kernel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/05motr-single-node/07multi-client-tests target=_blank>05motr-single-node/07multi-client-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/05motr-single-node/12sns-cc-io-tests target=_blank>05motr-single-node/12sns-cc-io-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/05motr-single-node/39mount-fail target=_blank>05motr-single-node/39mount-fail</a></p></details></td></tr><tr><td>:heavy_check_mark:Passed</td><td>57</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/01motr-single-node/01kernel-tests target=_blank>01motr-single-node/01kernel-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/01motr-single-node/03initscript-tests target=_blank>01motr-single-node/03initscript-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/01motr-single-node/04net-tests target=_blank>01motr-single-node/04net-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/01motr-single-node/05rpcping-tests target=_blank>01motr-single-node/05rpcping-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/01motr-single-node/06console-tests target=_blank>01motr-single-node/06console-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/01motr-single-node/28motr-sys-kvs-test target=_blank>01motr-single-node/28motr-sys-kvs-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/01motr-single-node/29confgen-test target=_blank>01motr-single-node/29confgen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/01motr-single-node/30hagen-test target=_blank>01motr-single-node/30hagen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/01motr-single-node/35m0singlenode target=_blank>01motr-single-node/35m0singlenode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/01motr-single-node/46m0crate target=_blank>01motr-single-node/46m0crate</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/01motr-single-node/52singlenode-sanity target=_blank>01motr-single-node/52singlenode-sanity</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/02motr-single-node/02system-tests target=_blank>02motr-single-node/02system-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/02motr-single-node/08degraded-mode-tests target=_blank>02motr-single-node/08degraded-mode-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/02motr-single-node/09poolmach-tests target=_blank>02motr-single-node/09poolmach-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/02motr-single-node/20pool-version-test target=_blank>02motr-single-node/20pool-version-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/02motr-single-node/21spiel-test target=_blank>02motr-single-node/21spiel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/02motr-single-node/22rpc-cancel-test target=_blank>02motr-single-node/22rpc-cancel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/02motr-single-node/23m0d-signal-test target=_blank>02motr-single-node/23m0d-signal-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/02motr-single-node/24m0d-fsync-test target=_blank>02motr-single-node/24m0d-fsync-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/02motr-single-node/41motr-conf-update target=_blank>02motr-single-node/41motr-conf-update</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/02motr-single-node/47motr-tests-user-kernel target=_blank>02motr-single-node/47motr-tests-user-kernel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/02motr-single-node/51kem target=_blank>02motr-single-node/51kem</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/02motr-single-node/53clusterusage-alert target=_blank>02motr-single-node/53clusterusage-alert</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/03motr-single-node/17sss-st-test target=_blank>03motr-single-node/17sss-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/03motr-single-node/36spare-reservation target=_blank>03motr-single-node/36spare-reservation</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/03motr-single-node/54sns-repair-motr-1f target=_blank>03motr-single-node/54sns-repair-motr-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/03motr-single-node/55sns-repair-motr-1n-1f target=_blank>03motr-single-node/55sns-repair-motr-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/03motr-single-node/56sns-repair-motr-mf target=_blank>03motr-single-node/56sns-repair-motr-mf</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/03motr-single-node/57sns-repair-motr-1k-1f target=_blank>03motr-single-node/57sns-repair-motr-1k-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/03motr-single-node/58sns-repair-motr-ios-fail target=_blank>03motr-single-node/58sns-repair-motr-ios-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/03motr-single-node/59sns-repair-motr-abort target=_blank>03motr-single-node/59sns-repair-motr-abort</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/03motr-single-node/60sns-repair-motr-abort-quiesce target=_blank>03motr-single-node/60sns-repair-motr-abort-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/03motr-single-node/61spiel-multi-confd target=_blank>03motr-single-node/61spiel-multi-confd</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/04motr-single-node/19sns-abort-test target=_blank>04motr-single-node/19sns-abort-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/04motr-single-node/25spiel-sns-repair-test target=_blank>04motr-single-node/25spiel-sns-repair-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/04motr-single-node/26spiel-sns-repair-quiesce-test target=_blank>04motr-single-node/26spiel-sns-repair-quiesce-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/04motr-single-node/27sns-repair-io-fail-test target=_blank>04motr-single-node/27sns-repair-io-fail-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/04motr-single-node/32m0t1fs-rconfc-fail-test target=_blank>04motr-single-node/32m0t1fs-rconfc-fail-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/04motr-single-node/33motr-st target=_blank>04motr-single-node/33motr-st</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/04motr-single-node/34sns-repair-1n-1f target=_blank>04motr-single-node/34sns-repair-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/04motr-single-node/48motr-raid0-io target=_blank>04motr-single-node/48motr-raid0-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/04motr-single-node/49motr-rpc-cancel target=_blank>04motr-single-node/49motr-rpc-cancel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/04motr-single-node/50motr-rm-lock-cc-io target=_blank>04motr-single-node/50motr-rm-lock-cc-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/04motr-single-node/51motr-rmw target=_blank>04motr-single-node/51motr-rmw</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/05motr-single-node/10sns-single-tests target=_blank>05motr-single-node/10sns-single-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/05motr-single-node/11sns-multi-tests target=_blank>05motr-single-node/11sns-multi-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/05motr-single-node/13sns-repair-quiesce target=_blank>05motr-single-node/13sns-repair-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/05motr-single-node/14m0t1fs-fsync-test target=_blank>05motr-single-node/14m0t1fs-fsync-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/05motr-single-node/15m0t1fs-fwait-test target=_blank>05motr-single-node/15m0t1fs-fwait-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/05motr-single-node/18m0mount-test target=_blank>05motr-single-node/18m0mount-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/05motr-single-node/38sns-abort-quiesce-test target=_blank>05motr-single-node/38sns-abort-quiesce-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/05motr-single-node/40motr-dgmode target=_blank>05motr-single-node/40motr-dgmode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/05motr-single-node/42motr-utils target=_blank>05motr-single-node/42motr-utils</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/05motr-single-node/43motr-sync-replication target=_blank>05motr-single-node/43motr-sync-replication</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/05motr-single-node/44motr-sns-repair target=_blank>05motr-single-node/44motr-sns-repair</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/05motr-single-node/45motr-sns-repair-N-1 target=_blank>05motr-single-node/45motr-sns-repair-N-1</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit/motr-single-node/62dix-repair-st-lookup-insert target=_blank>motr-single-node/62dix-repair-st-lookup-insert</a></p></details></td></tr><tr><td>Total</td><td>64</td><td><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/773//testReport/junit>:link:</a></td></tr></table><h2> CppCheck Summary</h2><p> &emsp;&emsp;&emsp;Cppcheck: No new warnings found :thumbsup:</p>

andriytk commented:
@mehjoshi, when merging - please squash all commits and use the PR description above as the commit msg. Thanks.
